File: Python/Add_Two_Numbers.py
class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        add_1 = False
        dummy_head = ListNode(0)
        curr = dummy_head
        while (l1 is not None) or (l2 is not None):
            x = l1.val if l1 is not None else 0
            y = l2.val if l2 is not None else 0
            sum = 0 if add_1 is False else 1
            sum += (x + y)
            if sum >= 10:
                sum = sum % 10
                add_1 = True
            else:
                add_1 = False
            l1 = l1.next if l1 is not None else None
            l2 = l2.next if l2 is not None else None
            curr.next = ListNode(sum)
            curr = curr.next
        if add_1:
            curr.next = ListNode(1)
        return dummy_head.next

File: C++/Add Two Numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* res=NULL;
        ListNode* temp=NULL;
        ListNode* prev=NULL;
        int carry=0;
        int sum;
        while(l1!=NULL || l2!=NULL){
            sum=carry+(l1?l1->val:0)+(l2?l2->val:0);
            carry=(sum>=10)?1:0;
            sum=sum%10;
            temp=new ListNode(sum);
            if(res==NULL)
                res=temp;
            else
                prev->next=temp;
            prev=temp;
            if(l1)
                l1=l1->next;
            if(l2)
                l2=l2->next;
        }
        if(carry>0)
            temp->next=new ListNode(carry);
        return res;
        
    }
};
